beacon-chain/sync/rpc_chunked_response.go

```diff
@@ -161,3 +161,98 @@ func WriteBlobSidecarChunk(stream libp2pcore.Stream, tor blockchain.TemporalOrac
 	_, err = encoding.EncodeWithMaxLength(stream, sidecar)
 	return err
 }
+
+func WriteLightClientBootstrapChunk(
+	stream libp2pcore.Stream,
+	tor blockchain.TemporalOracle,
+	encoding encoder.NetworkEncoding,
+	bootstrap interfaces.LightClientBootstrap,
+) error {
+	if _, err := stream.Write([]byte{responseCodeSuccess}); err != nil {
+		return err
+	}
+
+	valRoot := tor.GenesisValidatorsRoot()
+	digest, err := forks.ForkDigestFromEpoch(slots.ToEpoch(bootstrap.Header().Beacon().Slot), valRoot[:])
+	if err != nil {
+		return err
+	}
+	obtainedCtx := digest[:]
+
+	if err = writeContextToStream(obtainedCtx, stream); err != nil {
+		return err
+	}
+	_, err = encoding.EncodeWithMaxLength(stream, bootstrap)
+	return err
+}
+
+func WriteLightClientUpdateChunk(
+	stream libp2pcore.Stream,
+	tor blockchain.TemporalOracle,
+	encoding encoder.NetworkEncoding,
+	update interfaces.LightClientUpdate,
+) error {
+	if _, err := stream.Write([]byte{responseCodeSuccess}); err != nil {
+		return err
+	}
+
+	valRoot := tor.GenesisValidatorsRoot()
+	digest, err := forks.ForkDigestFromEpoch(slots.ToEpoch(update.AttestedHeader().Beacon().Slot), valRoot[:])
+	if err != nil {
+		return err
+	}
+	obtainedCtx := digest[:]
+
+	if err = writeContextToStream(obtainedCtx, stream); err != nil {
+		return err
+	}
+	_, err = encoding.EncodeWithMaxLength(stream, update)
+	return err
+}
+
+func WriteLightClientFinalityUpdateChunk(
+	stream libp2pcore.Stream,
+	tor blockchain.TemporalOracle,
+	encoding encoder.NetworkEncoding,
+	update interfaces.LightClientFinalityUpdate,
+) error {
+	if _, err := stream.Write([]byte{responseCodeSuccess}); err != nil {
+		return err
+	}
+
+	valRoot := tor.GenesisValidatorsRoot()
+	digest, err := forks.ForkDigestFromEpoch(slots.ToEpoch(update.AttestedHeader().Beacon().Slot), valRoot[:])
+	if err != nil {
+		return err
+	}
+	obtainedCtx := digest[:]
+
+	if err = writeContextToStream(obtainedCtx, stream); err != nil {
+		return err
+	}
+	_, err = encoding.EncodeWithMaxLength(stream, update)
+	return err
+}
+func WriteLightClientOptimisticUpdateChunk(
+	stream libp2pcore.Stream,
+	tor blockchain.TemporalOracle,
+	encoding encoder.NetworkEncoding,
+	update interfaces.LightClientOptimisticUpdate,
+) error {
+	if _, err := stream.Write([]byte{responseCodeSuccess}); err != nil {
+		return err
+	}
+
+	valRoot := tor.GenesisValidatorsRoot()
+	digest, err := forks.ForkDigestFromEpoch(slots.ToEpoch(update.AttestedHeader().Beacon().Slot), valRoot[:])
```

Comment:
We should use the update type to determine which fork version to use:
https://github.com/ethereum/consensus-specs/blob/dev/specs/electra/light-client/p2p-interface.md

Reply:
You are right, at the fork boundary we may attest to a header from the previous slot. But this poses a question if the type update itself can handle this because it only holds one type of header. I am not sure how this will work.

Reply:
this could be tested in e2e with an evaluator. Possibly having a very simple light client which just peers with our nodes and requests all the data each slot(bootstrap, finality update, optimistic update, etc)